test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/MockCollectorIntegrationTests.cs

```diff
@@ -91,6 +98,197 @@ public async Task TestRecoveryAfterFailedExport()
         await host.StopAsync();
     }
 
+    [Theory]
+    [InlineData(true, ExportResult.Success, Grpc.Core.StatusCode.Unavailable)]
```

Comment:
Add a test case for `StatusCode.DeadlineExceeded` as well.

Reply:
see https://github.com/open-telemetry/opentelemetry-dotnet/issues/5436. I added the comment

Reply:
We should still add a test case for it and expect the result to be `ExportResult.Failure` always?

Reply:
Added. It will be hard to show #5436 behavior here. The test would work as expected because there isn't actual delay in the request processing.

Reply:
We should be able to do it by tweaking the test to selectively return failure for both attempts, right? Something like this:

```csharp
var codes = new[] { initialStatusCode, Grpc.Core.StatusCode.OK };

if (initialStatusCode == StatusCode.DeadlineExceeded)
{
    codes[1] = initialStatusCode;
}
```

Anyway, I think if we decide that we would never retry on Deadline Exceeded, it's better to change the code to return `false` for that code instead of updating the test here.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/adc89d94249446cbd153b99330f274a76a33ca09/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/OtlpRetry.cs#L199-L207